packages/turbo-gen/src/utils/getWorkspaceStructure.ts

```diff
@@ -0,0 +1,56 @@
+import path from "path";
+import type { Project, Workspace } from "@turbo/workspaces";
+import { getWorkspaceRoots } from "./workspaceRoots";
+
+interface WorkspaceStructure {
+  hasRootApps: boolean;
+  hasRootPackages: boolean;
+  workspacesByGroup: Record<string, Array<Workspace>>;
+  nonAppWorkspaces: Array<Workspace>;
+}
+
+export function getGroupFromWorkspace({
+  project,
+  workspace,
+}: {
+  project: Project;
+  workspace: Workspace;
+}) {
+  return path
+    .relative(project.paths.root, workspace.paths.root)
+    .split(path.sep)[0];
+}
+
+export function getWorkspaceStructure({
+  project,
+}: {
+  project: Project;
+}): WorkspaceStructure {
+  // get the workspace roots first, any assumptions we make
+  // should at least be based around configured workspaces
+  const roots = getWorkspaceRoots({ project });
+  const hasRootApps = roots.includes("apps");
+  const hasRootPackages = roots.includes("packages");
```

Comment:
maybe pull these "known" directories into a constant somewhere so it's a little easier to track down/ customize if we need to. E.g. if someone has `sites/` for their apps